dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/program/DefaultProgramIndicatorService.java

```diff
@@ -202,6 +202,11 @@ public String getExpressionDescription(String expression) {
   @Override
   @Transactional(readOnly = true)
   public String getFilterDescription(String expression) {
+    return getFilterDescriptionWithoutTransaction(expression);
+  }
+
+  @Override
+  public String getFilterDescriptionWithoutTransaction(String expression) {
```

Comment:
The implementation makes me think that this does require a TX but what you want is to not open a new one in an already opened one? Or why would `getFilterDescription` open one if it is not needed?

Reply:
Removed. (See previous.)